fmc/src/flow/tci.rs

```diff
@@ -0,0 +1,57 @@
+/*++
+
+Licensed under the Apache-2.0 license.
+
+File Name:
+
+    tci.rs
+
+Abstract:
+
+    File contains execution routines for TCI computation
+
+Environment:
+
+    FMC
+
+
+--*/
+use crate::flow::crypto::Crypto;
+use crate::fmc_env::FmcEnv;
+use crate::HandOff;
+use caliptra_drivers::{Array4x12, CaliptraResult};
+use caliptra_image_types::ImageManifest;
+
+pub struct Tci {}
+
+impl Tci {
+    /// Compute Image Manifest Digest
+    ///
+    /// # Arguments
+    ///
+    /// * `env` - ROM Environment
+    /// * `hand_off` - HandOff
+    pub fn image_manifest_digest(
+        env: &mut FmcEnv,
+        hand_off: &HandOff,
+    ) -> CaliptraResult<Array4x12> {
+        let image_manifest_bytes = unsafe {
+            core::slice::from_raw_parts_mut(
+                hand_off.image_manifest_address(env) as *mut u8,
+                ImageManifest::default().size as usize,
```

Comment:
nit: Should this just be `sizeof::<ImageManifest>()`? I think you can avoid allocating the structure on the stack

Reply:
good catch. done